src/collectors/diskhealth/diskhealth.py

```diff
@@ -0,0 +1,150 @@
+# coding=utf-8
+
+"""
+Check Disk Health
+
+Note: The user running Diamond will need to have write access to each mount point
+that should be checked. The test performa a write and a read in each mount.
+
+#### Dependencies
+
+ * /proc/mount
+
+"""
+
+import diamond.convertor
+
+import time
+import os
+import re
+
+from multiprocessing import Pool, TimeoutError
+
+TEST_FILE_SIZE_BYTES = 1048576  # 1MB
+
+
+def _format_metric(device, mount_point, metric):
+    return '%s.%s.%s' % (
+        device.strip('/').replace('/', '_'),
+        'root' if mount_point == '/' else mount_point.strip('/').replace('/', '_'),
+        metric)
+
+
+def _test_disk(device, mount_point):
+    metrics = {}
+    filename = os.path.join(mount_point, '.diamond_diskhealth')
+
+    data = os.urandom(TEST_FILE_SIZE_BYTES)
+    start_time = time.time()
+    try:
+        f = open(filename, 'wb')
+        f.write(data)
+        f.flush()
+        f.close()
+        write_time = time.time()
```

Comment:
Yes, the read check makes sure that what you read is identically to what was written. if nothing was written then trying to read and compare it to what you were trying to write is not meaningful.